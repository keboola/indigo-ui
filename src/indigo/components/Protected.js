import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLock } from '@fortawesome/free-solid-svg-icons';

class Protected extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isProtected: true,
    };
  }

  render() {
    if (this.state.isProtected) {
      return this.renderProtected();
    }

    return <span>{this.props.children}</span>;
  }

  renderProtected() {
    return (
      <span title="Protected content, click to reveal">
        <FontAwesomeIcon
          icon={faLock}
          className="icon-fixed-width icon-clickable"
          onClick={() => this.setState({ isProtected: false })}
        />
      </span>
    );
  }
}

Protected.propTypes = {
  children: PropTypes.any,
};

export default Protected;
