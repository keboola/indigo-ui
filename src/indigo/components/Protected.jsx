import React from 'react';

export default React.createClass({
  getInitialState() {
    return {
      isProtected: true
    };
  },

  render() {
    return this.state.isProtected ? this.renderProtected() : this.renderRevealed();
  },

  renderProtected() {
    return (
      <span className="fa fa-fw fa-lock kbc-protected"
            title="Protected content, click to reveal"
            onClick={this.show} />
    );
  },

  renderRevealed() {
    return (
      <span>{this.props.children}</span>
    );
  },

  show() {
    this.setState({
      isProtected: false
    });
  }
});