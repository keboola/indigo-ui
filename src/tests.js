import renderer from 'react-test-renderer';

export const snapshot = (component) => {
  const tree = renderer
    .create(component)
    .toJSON();
  expect(tree).toMatchSnapshot();
};
