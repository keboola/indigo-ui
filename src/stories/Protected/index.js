import React from 'react';
import { storiesOf } from '@storybook/react';
import Protected from '../../indigo/components/Protected';

storiesOf('Protected', module).add(
  'Protected',
  () => {
    return <Protected>This is my secret</Protected>;
  },
  { info: { text: 'To hide protected content with option to show it.' } }
);
