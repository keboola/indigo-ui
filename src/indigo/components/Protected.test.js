import React from 'react';

import { snapshot } from "../../tests";
import Protected from './Protected';

describe('<Protected />', () => {
  
  it('Protected - content hidden (default)', () => {
    snapshot(<Protected>This is protected</Protected>);
  });

  // TODO: mock state?

});
