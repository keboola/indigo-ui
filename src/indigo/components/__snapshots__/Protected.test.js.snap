// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Protected /> Protected - content hidden (default) 1`] = `
<span
  title="Protected content, click to reveal"
>
  <svg
    aria-hidden="true"
    className="svg-inline--fa fa-lock fa-w-14 icon-fixed-width icon-clickable"
    data-icon="lock"
    data-prefix="fas"
    focusable="false"
    onClick={[Function]}
    role="img"
    style={Object {}}
    viewBox="0 0 448 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z"
      fill="currentColor"
      style={Object {}}
    />
  </svg>
</span>
`;
